### PR TITLE
Check the vote value is > Abstain in Governance vote function.

### LINKED ISSUE #6948: Governance vote accepts non-zero values which function as none instead of abstain
### Expected Behavior

Either `value>Abstain` should imply `value=Abstain` or revert

### Current Behavior

Governance `vote` reverts on `uint8 value=None` but not `uint8 value>Abstain`

### PR DESCRIPTION
**Pull request by deepakhb2**
### Description
Expected Behavior
Governance vote value > Yes and value <= None should be reverted

Current Behavior
Governance vote reverts on uint8 value=None but not uint8 value>Yes

### Other changes

_Describe any minor or "drive-by" changes here._

### Tested

Can't be unit tested as we can't pass negative values or positive values greater than Yes vote is not allowed. 

### Related issues

- Fixes #6948

### Backwards compatibility

It doesn't change any existing feature except handling the boundary condition.

### Documentation

N/A

### CODE REVIEW

**Comment by deepakhb2** [comment]
@yorhodes @nambrot Now the Abstain votes are not allowed and 4 tests are failing due to this change. Please review the change. Thank you.

File: packages/protocol/contracts/governance/Governance.sol [hunk 1]
```diff
@@ -629,7 +629,7 @@ contract Governance is
     uint256 weight = getLockedGold().getAccountTotalLockedGold(account);
     require(proposal.isApproved(), "Proposal not approved");
     require(stage == Proposals.Stage.Referendum, "Incorrect proposal state");
-    require(value != Proposals.VoteValue.None, "Vote value unset");
+    require(value > Proposals.VoteValue.Abstain, "Vote value unset");
```
Comment (yorhodes) [inline packages/protocol/contracts/governance/Governance.sol]:
```suggestion
    require(value <= Proposals.VoteValue.Abstain, "Vote value unset");
```

File: packages/protocol/test/governance/network/governance.ts [hunk 2]
```diff
@@ -1566,6 +1566,10 @@ contract('Governance', (accounts: string[]) => {
       })
     })
 
+    it('should revert for abstain vote', async () => {
+      await assertRevert(governance.vote(proposalId, index, VoteValue.Abstain))
```
Comment (yorhodes) [inline packages/protocol/test/governance/network/governance.ts]:
```suggestion
      await assertRevert(governance.vote(proposalId, index, VoteValue.Abstain + 1))
```

**Review by yorhodes — CHANGES_REQUESTED** [review]

File: packages/protocol/contracts/governance/Governance.sol (hunk 1 above)
Comment (deepakhb2) [reply packages/protocol/contracts/governance/Governance.sol]: I am bit confused. I will make this change and see how this works. Now, I understand the reason for this change. Thank you.

Comment (deepakhb2) [reply packages/protocol/contracts/governance/Governance.sol]: @yorhodes This change will fail most of the test cases related to vote method as 'yes' and 'no' are greater then 'abstain'. I am not clear on this change. Please confirm if this change is correct. Thank you.

File: packages/protocol/test/governance/network/governance.ts [hunk 3]
```diff
@@ -1575,10 +1575,6 @@ contract('Governance', (accounts: string[]) => {
       })
     })
 
-    it('should revert for abstain plus 1 vote', async () => {
-      await assertRevert(governance.vote(proposalId, index, VoteValue.Abstain + 1))
-    })
-
```
Comment (yorhodes) [inline packages/protocol/test/governance/network/governance.ts]: can we make this revert on `VoteValue.Yes + 1` now? 
sorry again for the confusion here

Comment (deepakhb2) [reply packages/protocol/test/governance/network/governance.ts]: I tried and it works for Abstain as in enum has next value as No. But, for Yes is the last value in enum so it doesn't allow and will raise an exception.
`errored: VM error: invalid opcode.`

Comment (yorhodes) [reply packages/protocol/test/governance/network/governance.ts]: wait what?!?!
so we didn't need to `require` then?

Comment (deepakhb2) [reply packages/protocol/test/governance/network/governance.ts]: Yes, it is not required. Also it will take additional gas to check two conditions.

Comment (yorhodes) [reply packages/protocol/test/governance/network/governance.ts]: very true
tbh I thought the compiler treated enums as uints
I guess we can close this one then?
cc @nambrot

Comment (deepakhb2) [reply packages/protocol/test/governance/network/governance.ts]: Yes we can close this PR and issue as well.
